File: c/roman_to_integer.c
#include <stdio.h>
#include <string.h>

int numerals(char character)
{
	switch(character)
	{
		case 'I' : return 1;
		case 'V' : return 5;
		case 'X' : return 10;
		case 'L' : return 50;
		case 'C' : return 100;
		case 'D' : return 500;
		case 'M' : return 1000;
	}

	return 0;
}

int roman_to_integer(char* string)
{
	int integer = 0;

	for(int i = 0; i < strlen(string); i++)
	{
		if(i + 1 >= strlen(string))
		{
			integer += numerals(string[i]);
		}

		if(i + 1 < strlen(string))
		{
			if(numerals(string[i]) >= numerals(string[i + 1])) integer += numerals(string[i]);
			if(numerals(string[i]) < numerals(string[i + 1])) {integer += numerals(string[i + 1]) - numerals(string[i]); i++;}
		}
	}

	return integer;
}

char* string_1 = "IV";
char* string_2 = "IX";
char* string_3 = "III";
char* string_4 = "LVIII";
char* string_5 = "MMCXVI";
char* string_6 = "MCMXCIV";

int main()
{
	printf("%d\n", roman_to_integer(string_1));
	printf("%d\n", roman_to_integer(string_2));
	printf("%d\n", roman_to_integer(string_3));
	printf("%d\n", roman_to_integer(string_4));
	printf("%d\n", roman_to_integer(string_5));
	printf("%d\n", roman_to_integer(string_6));
}

File: c#/RomanToInteger.cs
using System;
using System.Collections.Generic;

public class Solution
{
	public int RomanToInteger(string str)
	{
		int integer = 0;
		Dictionary<char, int> numerals = new Dictionary<char, int>();

		numerals.Add('I', 1);
		numerals.Add('V', 5);
		numerals.Add('X', 10);
		numerals.Add('L', 50);
		numerals.Add('C', 100);
		numerals.Add('D', 500);
		numerals.Add('M', 1000);

		for(int i = 0; i < str.Length; i++)
		{
			if(i + 1 >= str.Length)
			{
				integer += numerals[str[i]];
			}

			if(i + 1 < str.Length)
			{
				if(numerals[str[i]] >= numerals[str[i + 1]]) integer += numerals[str[i]];
				if(numerals[str[i]] < numerals[str[i + 1]]) {integer += numerals[str[i + 1]] - numerals[str[i]]; i++;}
			}
		}

		return integer;
	}
}

public class LeetCode
{
	public static void Main()
	{
		string string1 = "IV";
		string string2 = "IX";
		string string3 = "III";
		string string4 = "LVIII";
		string string5 = "MMCXVI";
		string string6 = "MCMXCIV";
		Solution solution = new Solution();

		Console.WriteLine(solution.RomanToInteger(string1));
		Console.WriteLine(solution.RomanToInteger(string2));
		Console.WriteLine(solution.RomanToInteger(string3));
		Console.WriteLine(solution.RomanToInteger(string4));
		Console.WriteLine(solution.RomanToInteger(string5));
		Console.WriteLine(solution.RomanToInteger(string6));
	}
}